OpenSimRT/Common/include/OpenSimUtils.h

```diff
@@ -74,27 +74,27 @@ struct Common_API OpenSimUtils {
 
     /**
      * Compare two OpenSim::Datatables or any of the derived types (as long as
-     * they share the same data types). Number of columns and rows must match,
-     * otherwise throws an exception. Comparison is performed by computing the
-     * RMSE of the columns with common labels acquired from the 'query' table.
-     * If the computed RMSE of any of the match columns is larger than the given
-     * threshold it throws an exception.
+     * they share the same data types). Comparison is performed by computing the
+     * RMSE of the columns with common labels acquired from the 'query' table
+     * (default), or by computing the RMSE of the rows with common timestamp
+     * values. The column labels of the query table must exist in the reference
+     * table, otherwise it throws exception. If columns are compared, the number
+     * of rows must also match. If the computed RMSE of any of the match
+     * columns/rows is larger than the given threshold it throws an exception.
      *
      * @param queryTable - Query table to compare.
      *
      * @param refTable - Reference table to be compare against.
      *
      * @param threshold - Threshold value (default is 1e-5).
+     *
+     * @param compareColumns - Choose to compare table columns or rows.
      */
     template <template <typename, typename> class Table, typename T, typename E>
     static void compareTables(const Table<T, E>& queryTable,
                               const Table<T, E>& refTable,
-                              const double& threshold = 1e-5) {
-        if (!(queryTable.getNumRows() == refTable.getNumRows()))
-            THROW_EXCEPTION("Number of rows in given tables do not match.");
-        if (!(queryTable.getNumColumns() == refTable.getNumColumns()))
-            THROW_EXCEPTION("Number of columns in given tables do not match.");
-
+                              const double& threshold = 1e-5,
+                              const bool& compareColumns = true) {
```

Comment:
Should verify that previous test work properly.

Reply:
Tests work and pass as expected.